sql/src/main/java/io/crate/execution/engine/collect/collectors/OrderedLuceneBatchIteratorFactory.java

```diff
@@ -126,7 +135,12 @@ private void onNextRows(List<KeyIterable<ShardId, Row>> rowsList, @Nullable Thro
         }
 
         private void mergeAndMaybeFinish(List<KeyIterable<ShardId, Row>> rowsList) {
-            pagingIterator.merge(rowsList);
+            try {
+                pagingIterator.merge(rowsList);
+            } catch (Throwable t) {
+                batchPagingIterator.kill(t);
```

Comment:
why do you need the kill(t) call?
it just assigns the exception to `kill` variable and is not used by close().